Reflection/CustomConfigurationDemo/ConfigurationComponentBase.cs

```diff
@@ -0,0 +1,77 @@
+﻿using System.Reflection;
+using System.Runtime.CompilerServices;
+using CustomConfigurationDemo.Attributes;
+using CustomConfigurationDemo.Providers;
+
+namespace CustomConfigurationDemo;
+
+public abstract class ConfigurationComponentBase
+{
+    protected TValue? GetPropertyValue<TValue>([CallerMemberName] string? propertyName = null)
```

Comment:
Well, not sure that `string?` is a valid since it's a reference type and its initial value is `null`. Can we leave just `string` as in `SetPropertyValue`?